tests/main/snapctl/task.yaml

```diff
@@ -44,7 +44,7 @@ execute: |
         exit 1
     fi
 
-    if ! printf 'GET /v2/snaps HTTP/1.0\r\n\r\n' | nc -U -w 1 /run/snapd-snap.socket | grep '401 Unauthorized'; then
-        echo "Expected snapd API to be unauthorized on the snap socket"
+    if ! printf 'GET /v2/snaps HTTP/1.0\r\n\r\n' | nc -U -w 1 /run/snapd-snap.socket | grep '403 Forbidden'; then
+        echo "Expected snapd API to be forbidden on the snap socket"
```

Comment:
With the existing access check code, it looks like you would be able to access the full API via `snapd-snap.socket` if you provide a macaroon.  However, it would not be possible to obtain a macaroon from `/v2/login` from that socket.

It's not clear that this is desirable behaviour, even if some snap has managed it somehow (using `personal-files` to access `~/.snap`? somehow finding one via `password-manager-service`?).